### PR TITLE
Change auth flow to use Bot API

### LINKED ISSUE #9: Token -> acess_token
Instead storing the token directly by doing a get request to the go auth service we built we can receive a session id which can be used to access more stuff as it has access to the bot.

### PR DESCRIPTION
**Pull request by RealYusufIsmail**

### CODE REVIEW

File: src/utils/api.ts [hunk 1]
```diff
@@ -15,13 +17,44 @@ export const getAuthLogin = () => {
 
 export function handleGuild(json: any): Guild | null {
     try {
-        if (json.permissions & 0x0000000000000020) {
-            return new GuildImpl(json.id, json)
-        } else {
-            return null
-        }
+        return new GuildImpl(json.id, json)
```
Comment (RealYusufIsmail) [inline src/utils/api.ts:20]: Have a look it. Might need to add back

Comment (RealYusufIsmail) [reply src/utils/api.ts:20]: Nvm

File: src/App.tsx [hunk 2]
```diff
@@ -19,12 +19,12 @@ function App() {
     //state variable and function to update the state
     const updateGuild = (guild: Guild) => setGuild(guild);
 
-    //if token exists in cookies, redirect to dashboard
+    //if acccess token exists in cookies, redirect to dashboard
     //if not, redirect to login page
 
-    const token = getCookie("token")
+    const accesToken = getCookie("access_token")
```
Comment (SIMULATAN) [inline src/App.tsx]: Typo: change to "accessToken"

File: src/App.tsx [hunk 3]
```diff
@@ -19,12 +19,12 @@ function App() {
     //state variable and function to update the state
     const updateGuild = (guild: Guild) => setGuild(guild);
 
-    //if token exists in cookies, redirect to dashboard
+    //if acccess token exists in cookies, redirect to dashboard
```
Comment (SIMULATAN) [inline src/App.tsx]: Typo: change to "access"

File: README.md [hunk 4]
```diff
@@ -33,6 +33,7 @@ To run the app you need to create a config.json file in the src/security folder
   "client_id": "",
   "client_secret": "",
   "redirect_url": "",
-  "auth_api_url": ""
+  "auth_api_url": "",
```
Comment (SIMULATAN) [inline README.md]: `auth_api_url` was changed to `bot_api`

File: README.md [hunk 5]
```diff
@@ -33,6 +33,7 @@ To run the app you need to create a config.json file in the src/security folder
   "client_id": "",
   "client_secret": "",
```
Comment (SIMULATAN) [inline README.md]: To be removed, the secret is only required in the backend.
In fact, when it's supplied, the client will get it and compromise our OAuth2

File: src/pages/onboarding/Callback.tsx [hunk 6]
```diff
@@ -24,31 +24,18 @@ export const CallbackPage = () => {
...
                 window.location.href = "/";
                 return;
             }
 
-            //raw expires_in is 604800 for example, which is 7 days
-            let formatExperiesIn = new Date();
-            formatExperiesIn.setTime(formatExperiesIn.getTime() + (json.expires_in * 1000));
-
-            setCookie("token", json.access_token, {path: "/", expires: formatExperiesIn});
+            //todo : change sent text to include expiry time
```
Comment (SIMULATAN) [inline src/pages/onboarding/Callback.tsx]: Implement this
The JWT token can be decoded to get the expiration time.
Refreshing that token is out of scope for this PR.

**Review by SIMULATAN — CHANGES_REQUESTED** [review]
Some minor changes are required, other than those comments, LGTM!

File: src/pages/onboarding/Callback.tsx (hunk 6 above)
Comment (RealYusufIsmail) [reply src/pages/onboarding/Callback.tsx]: What do you send, so i know what to decode

Comment (SIMULATAN) [reply src/pages/onboarding/Callback.tsx]: One example token issued for my user:
```
eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2ODE5Mzk4OTUsImlzcyI6IkJsYWNrT25pb24tQm90Iiwic3ViIjoiNTA1NzEzNzYwMTI0NjY1ODY3In0.QN4V7ueA4lMItx2PJEeIJyBYgBRoFK5maGUj5KAN7fXSYmqoHDbKFHiVCS2vR6IsV3V1FRSXek-_ZdfGfyGJoA
```
and before anyone asks, no, you can't use that token anywhere.

Anyway, using [jwt.io](https://jwt.io/#debugger-io?token=eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2ODE5Mzk4OTUsImlzcyI6IkJsYWNrT25pb24tQm90Iiwic3ViIjoiNTA1NzEzNzYwMTI0NjY1ODY3In0.QN4V7ueA4lMItx2PJEeIJyBYgBRoFK5maGUj5KAN7fXSYmqoHDbKFHiVCS2vR6IsV3V1FRSXek-_ZdfGfyGJoA), you can check the token and see the information stored in it. According to the JWT specification, you can literally just Base64 decode the 2nd part of the token to get the JSON stored in it. In there, you can see the `exp` date which is the unix timestamp (seconds) at which the token expires.

In the future, you'll have to refresh that token every `MAX_LIFETIME / 2`. In this case, the lifetime is 7 days, as seen [here](https://github.com/Black0nion/BlackOnion-Bot/blob/c0a628e20bb38962b43f2b8e6cd035b6f88862e6/src/main/java/com/github/black0nion/blackonionbot/utils/AuthUtils.java#L28), so you'll have to refresh the token roughly every 4-5 days, but as I said, that's for another PR.

TL;DR: base64 decode the 2nd out of 3 parts of the token the backend provides you and get the `exp` json field of it to check at which unix epoch timestamp it expires.

File: src/pages/onboarding/Callback.tsx [hunk 7]
```diff
@@ -24,31 +24,23 @@ export const CallbackPage = () => {
...
-            let formatExperiesIn = new Date();
-            formatExperiesIn.setTime(formatExperiesIn.getTime() + (json.expires_in * 1000));
+            //base64 decode the 2nd out of 3 parts of the token the backend provides you and get the exp json field of it to check at which unix epoch timestamp it expires.
+            const decodedToken = atob(accessToken.split(".")[1]);
+            const decodedTokenJson = JSON.parse(decodedToken);
+            const expiryTime = decodedTokenJson.exp;
+            const formattedExpiryTime = new Date(expiryTime * 1000);
 
-            setCookie("token", json.access_token, {path: "/", expires: formatExperiesIn});
+            setCookie("access_token", accessToken, {path: "/", expires: formattedExpiryTime});
```
Comment (SIMULATAN) [inline src/pages/onboarding/Callback.tsx:43]: Oh wow, that exists? Great!

**Review by SIMULATAN — APPROVED** [review]
LGTM!

File: src/pages/onboarding/Callback.tsx (hunk 7 above)
Comment (RealYusufIsmail) [reply src/pages/onboarding/Callback.tsx:43]: No problem
